homeassistant/components/amcrest/camera.py

```diff
@@ -392,33 +387,44 @@ def update(self) -> None:
         _LOGGER.debug("Updating %s camera", self.name)
         try:
             if self._brand is None:
-                resp = self._api.vendor_information.strip()
+                resp = await self._api.async_vendor_information
                 _LOGGER.debug("Assigned brand=%s", resp)
                 if resp:
                     self._brand = resp
                 else:
                     self._brand = "unknown"
             if self._model is None:
-                resp = self._api.device_type.strip()
+                resp = await self._api.async_device_type
                 _LOGGER.debug("Assigned model=%s", resp)
                 if resp:
                     self._model = resp
                 else:
                     self._model = "unknown"
             if self._attr_unique_id is None:
-                serial_number = self._api.serial_number.strip()
+                serial_number = (await self._api.async_serial_number).strip()
                 if serial_number:
                     self._attr_unique_id = (
                         f"{serial_number}-{self._resolution}-{self._channel}"
                     )
                     _LOGGER.debug("Assigned unique_id=%s", self._attr_unique_id)
-            self._attr_is_streaming = self._get_video()
-            self._is_recording = self._get_recording()
-            self._motion_detection_enabled = self._get_motion_detection()
-            self._audio_enabled = self._get_audio()
-            self._motion_recording_enabled = self._get_motion_recording()
-            self._color_bw = self._get_color_mode()
-            self._rtsp_url = self._api.rtsp_url(typeno=self._resolution)
+            if self._rtsp_url is None:
+                self._rtsp_url = await self._api.async_rtsp_url(typeno=self._resolution)
```

Comment:
Can this change?

Reply:
No, the resolution (really this is which stream to use, main vs substream) is a part of the configuration for the camera.